File: src/main/java/hwr/oop/examples/polymorphism/subtyp/Quadrangle.java
package hwr.oop.examples.polymorphism.subtyp;

public interface Quadrangle extends Shape, Comparable<Quadrangle> {
    @Override
    default int getNumberOfSides() { return 4; }

    double[] getDistinctAngles();

    @Override
    default int compareTo(Quadrangle o) {
        return Double.compare(getAreaSize(), o.getAreaSize());
    }
}

File: examples-kotlin/lecture-05/src/main/kotlin/hwr/oop/examples/abstraction/realization/Quadrangle.kt
package hwr.oop.examples.abstraction.realization

interface Quadrangle : Shape, Comparable<Quadrangle> {

  override fun numberOfSides(): Int = 4

  fun distinctAngles(): Set<Double>

  override fun compareTo(other: Quadrangle): Int {
    return areaSize().compareTo(other.areaSize())
  }

}
